modules/core/02-client/migrations/v7/store.go

```diff
@@ -150,28 +133,14 @@ func handleLocalhostMigration(ctx sdk.Context, store sdk.KVStore, cdc codec.Bina
 // avoid state corruption as modifying state during iteration is unsafe. A special case
 // for tendermint clients is included as only one tendermint clientID is required for
 // v7 migrations.
-func collectClients(ctx sdk.Context, store sdk.KVStore, clientType string) (clients []string, err error) {
-	clientPrefix := []byte(fmt.Sprintf("%s/%s", host.KeyClientStorePrefix, clientType))
-	iterator := sdk.KVStorePrefixIterator(store, clientPrefix)
-
-	defer iterator.Close()
-	for ; iterator.Valid(); iterator.Next() {
-		path := string(iterator.Key())
-		if !strings.Contains(path, host.KeyClientState) {
-			// skip non client state keys
-			continue
-		}
-
-		clientID := host.MustParseClientStatePath(path)
-		clients = append(clients, clientID)
-
-		// optimization: exit after a single tendermint client iteration
-		if strings.Contains(clientID, exported.Tendermint) {
-			return clients, nil
-		}
-	}
-
-	return clients, nil
+func collectClients(ctx sdk.Context, keeper ClientKeeper, clientType string) (clients []string, err error) {
+	var clientIDs []string
+	keeper.IterateClientStates(ctx, nil, func(clientID string, _ exported.ClientState) bool {
+		clientIDs = append(clientIDs, clientID)
+		return false
+	})
+
+	return clientIDs, nil
 }
 
 // removeAllClientConsensusStates removes all client consensus states from the associated
```

Comment:
This function can be improved if we add a prefix to the `IterateConsensusStates` function